localization/ekf_localizer/src/covariance.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ekf_localizer/covariance.hpp"
+
+std::array<double, 36> ekfCovarianceToPoseMessageCovariance(const Matrix6d & P)
+{
+  const double p00 = P(0, 0);
```

Comment:
Why not using `IDX::X` or `COV_IDX::X_X` type of expressions here, instead of hard-coding the indices like `(0, 0)`?

```suggestion
  std::array<double, 36> p;
  p[COV_IDX::X_X] = P(IDX::X, IDX::X);
```

Reply:
Because I thought it would be easier to see the matrix by writing like following

```cpp
p00, p01, 0.0, 0.0, 0.0, p02,
p10, p11, 0.0, 0.0, 0.0, p12,
0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 
0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 
p20, p21, 0.0, 0.0, 0.0, p22
``` 

The bot automatically reformatted though

Reply:
And when we really need to modify the covariance matrix, the way of writing using the array is not very intuitive in my opinion

Reply:
Leaving `tier4_autoware_utils::xyzrpy_covariance_index` aside, I believe that it is better to use `IDX::X, IDX::X` type of indices for the input covariance matrix.

Considering the case in which the EKF state changes (e.g. [x,y,z,yaw,roll,pitch,...]), using the above indices would be more flexible IMO.

Reply:
How about https://github.com/autowarefoundation/autoware.universe/pull/2018/commits/bb22c3c40b0591e29adc01d420d1e0cd9b3071cd?

Reply:
LGTM!